### ISSUE TITLE
Opening files in Google Drive folders does not work

### ISSUE THREAD

**Issue by lplatter** [rationalmatter/Juno-Issue-Tracker#96]
**Describe the bug**
I cannot open existing jupyter notebooks that are in Google Drive Folders. I also cannot create new notebooks inside Google Drive folders.

**To Reproduce**
1. Go to Folders
2. Existing notebooks are greyed out.
3. Tapping on "New Notebook" gives message "Failed to open notebook".
**Expected behavior**
I expect to open a notebook inside a Google Drive folder.

**Environment (please complete the following information):**
 - Device: Ipad 6th generation
 - iOS version: 13.2.4
 - App version , 1.1.1

**Comment by navoshta** [comment]
@lplatter Yep Google Drive is not playing well with the Files app integration, and, for example, doesn't allow granting access to folders — I _think_ they previously allowed creating documents at least, and this might have changed in iOS 13. I'll see if there is anything that I could do in Juno, but IIRC it's something that they need to allow in their app, in the way their document provider extension is implemented.

**Comment by lplatter** [comment]
On my Ipad, Notability syncs to Google Drive. Notability creates files and folders within its Google Drive folder.

**Comment by navoshta** [comment]
@lplatter Notability is integrating directly with Google Drive, whereas Juno integrates with system's Files app instead. This is supposed to be a better way of file system integration, as (1) you get consistent system UI for browsing files in general, and (2) you no longer need to integrate with every possible cloud storage provider — instead, each storage provider integrates with Files. Unfortunately, not all cloud storage providers choose to integrate everything iOS's Files app has to offer (they must have their reasons I suppose).

**Comment by carlossierra311** [comment]
I have the same problem with Juno on Google Drive and a similar problem on Dropbox, but the problem there is that my notebook will open, but a csv file referenced from, it won’t. When I store everything locally on my iPad, the notebook works without issues, opening the reference file properly. See this video for evidence of everything I described here: 
https://photos.app.goo.gl/3RKrVprq54DRNquf7

**Comment by navoshta** [comment]
@carlossierra311 Yes, unfortunately this is due to Google Drive and Dropbox choosing _not_ to integrate with the iOS Files app. Not much Juno can do on its side without them integrating from their end first. You can give it a go with iCloud, or Working Copy app (git client for iOS), which should both work perfectly and let you grant access to your notebook's folder in its sandbox, as well as edit all folder contents in place.

Few notes on your video: please, keep in mind that if you see a "locked folder" icon in the notebook editor (pulsating in your video), Juno won't have access to notebook folder by default, you need to explicitly grant access to it (by tapping said lock icon). Also, keep in mind that in order for Juno to be able to open a file stored in the cloud (e.g. in iCloud), you do need to download it to device first. Not that any of this would help here, as neither Dropbox nor Google Drive allow folder access in their Files integrations.

**Comment by carlossierra311** [comment]
Thanks @navoshta. I will switch to Working Copy, which I already use for other projects.

**Comment by manentai** [comment]
Let’s hope this integration from Google and Dropbox will happen at some point and I hope you will let your users know. Thanks for your explanation @navoshta
